x-pack/plugins/cases/server/services/user_actions/index.ts

```diff
@@ -434,6 +435,102 @@ export class CaseUserActionService {
       throw error;
     }
   }
+
+  public async getUniqueConnectorOfCase({
+    caseId,
+    filter,
+    unsecuredSavedObjectsClient,
+  }: {
+    caseId: string;
+    unsecuredSavedObjectsClient: SavedObjectsClientContract;
+    filter?: KueryNode;
+  }): Promise<Array<{ id: string }>> {
+    try {
+      this.log.debug(`Attempting to count connectors for case id ${caseId}`);
+      const connectorsFilter = buildFilter({
+        filters: [ActionTypes.connector, ActionTypes.create_case],
+        field: 'type',
+        operator: 'or',
+        type: CASE_USER_ACTION_SAVED_OBJECT,
+      });
+
+      const combinedFilter = combineFilters([connectorsFilter, filter]);
+
+      const response = await unsecuredSavedObjectsClient.find<
+        CaseUserActionAttributesWithoutConnectorId,
+        { references: { connectors: { ids: { buckets: Array<{ key: string }> } } } }
+      >({
+        type: CASE_USER_ACTION_SAVED_OBJECT,
+        hasReference: { type: CASE_SAVED_OBJECT, id: caseId },
+        page: 1,
+        perPage: 1,
+        sortField: defaultSortField,
+        aggs: this.buildCountConnectorsAggs(),
+        filter: combinedFilter,
+      });
+
+      return (
+        response.aggregations?.references?.connectors?.ids?.buckets?.map(({ key }) => ({
+          id: key,
+        })) ?? []
+      );
+    } catch (error) {
+      this.log.error(`Error while counting connectors for case id ${caseId}: ${error}`);
+      throw error;
+    }
+  }
+
+  private buildCountConnectorsAggs(
+    /**
+     * It is high unlikely for a user to have more than
+     * 100 connectors attached to a case
+     */
+    size: number = 100
+  ): Record<string, estypes.AggregationsAggregationContainer> {
+    return {
+      references: {
+        nested: {
+          path: `${CASE_USER_ACTION_SAVED_OBJECT}.references`,
+        },
+        aggregations: {
+          connectors: {
+            filter: {
+              term: {
+                [`${CASE_USER_ACTION_SAVED_OBJECT}.references.type`]: 'action',
+              },
+            },
+            aggregations: {
+              ids: {
+                terms: {
+                  field: `${CASE_USER_ACTION_SAVED_OBJECT}.references.id`,
+                  size,
+                },
+                aggregations: {
+                  docs: {
```

Comment:
Doesn't look like we're using the result from this in the `getUniqueConnectorOfCase` method. Can we remove this part of the aggregation? Or is the plan to use it in another method?

Reply:
Hey! I plan to use it in the future when we want to get the fields of each connector attached to a case and remove the burden from the front end.

Reply:
I removed it.